File: 프로그래머스/lv1/leehandsub/가장 가까운 같은 글자.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

vector<int> solution(string s) {
    vector<int> answer;
    map<char,int> m;
    
    for(int i=0;i<s.size();i++)
    {
        if(m.count(s[i])==1)
        {
            answer.push_back(i-m[s[i]]);
            m[s[i]]=i;
        }
        else
        {
            m.insert({s[i],i});
            answer.push_back(-1);
        }
    }
    
    
    return answer;
}

File: 프로그래머스/lv1/이재표/가장 가까운 같은 글자.java
import java.util.*;;
class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        Map<Character,Integer>map=new HashMap<>();
        for(int i=0;i<s.length();i++){
            int idx=map.getOrDefault(s.charAt(i),-1);
            if(idx==-1){
                answer[i]=-1;
            }else{
                answer[i]=i-idx;
            }
            map.put(s.charAt(i),i);
        }
        return answer;
        
    }
}
